signac/core/dict_manager.py

```diff
@@ -108,7 +108,7 @@ def __delattr__(self, name):
 
     def __iter__(self):
         for fn in os.listdir(self.prefix):
-            m = re.fullmatch('(.*){}'.format(self.suffix), fn)
+            m = re.match('^(.*){}$'.format(self.suffix), fn)
```

Comment:
I think using the `^(.*){}$` syntax is much nicer.

Reply:
As @bdice said. Good choice.

Reply:
It's also necessary as long as we support Python 2.7, since `re.fullmatch()` is Py3 only.

Reply:
Ah, I am not familiar with that, though I assumed something like that with the commit text. Regardless, I also think it is more clear about the purpose of the code whether for python 2 or 3.